File: src-tauri/src/commands/external/ffz.rs
use crate::dto::external::emote::EmoteEntry;
use crate::services;

#[tauri::command]
pub async fn ffz_get_global_emotes() -> Result<Vec<EmoteEntry>, String> {
    services::external::ffz::get_global_emotes().await
}

#[tauri::command]
pub async fn ffz_get_channel_emotes(channel_login: String) -> Result<Vec<EmoteEntry>, String> {
    services::external::ffz::get_channel_emotes(channel_login).await
}

File: src/lib/api/external/ffz.ts
import type { EmoteEntry } from "../../types/external/emote.ts";
import { invokeCommand, type InvokeOptions } from "../utils.ts";

export function ffzGetGlobalEmotes(
  options?: InvokeOptions,
): Promise<EmoteEntry[]> {
  return invokeCommand("ffz_get_global_emotes", undefined, options);
}

export type FfzGetChannelEmotesParams = {
  channelLogin: string;
};

export function ffzGetChannelEmotes(
  params: FfzGetChannelEmotesParams,
  options?: InvokeOptions,
): Promise<EmoteEntry[]> {
  return invokeCommand("ffz_get_channel_emotes", params, options);
}
